src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
                 {
                     statements.Add(F.Assignment(F.Field(F.This(), field), F.Field(param, field)));
                 }
             }
         }
+
+        internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType)
+        {
+            foreach (var member in containingType.InstanceConstructors)
+            {
+                if (member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
+                    method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
+                    method.Parameters[0].RefKind == RefKind.None)
+                {
+                    return method;
```

Comment:
> return method; [](start = 19, length = 15)

Also, it is quite possible to have two constructors in metadata different only by custom modifiers, I think we shouldn't just pick one of them arbitrarily and should report an error instead. Please add a test.

Reply:
We might want to try to disambiguate based on the number of custom modifiers, as overload resolution does, I believe. However, there is still no guarantee that that would help in all cases.

---
In reply to: [438509983](https://github.com/dotnet/roslyn/pull/45022#discussion_r438509983) [](ancestors = 438509983)